[Bug]: Candle Notional Migration Script still causing deadlock in CI

### Problem encountered

[Bug]: Candle Notional Migration Script still causing intermittent deadlock in CI

### Observed behaviour

from https://jenkins.ops.vega.xyz/blue/rest/organizations/jenkins/pipelines/vega/branches/PR-8820/runs/5/log/?start=0


2023-07-18T08:47:43.907Z	INFO	revert-schema-to-version-0	v3@v3.6.1/migration.go:38	OK   0010_stop_orders.sql
2023-07-18T08:47:43.911Z	INFO	revert-schema-to-version-0	v3@v3.6.1/migration.go:38	OK   0011_gov_proposals2.sql
    service_test.go:1037: 
        	Error Trace:	/jenkins/workspace/vega_PR-8820/vega/datanode/networkhistory/service_test.go:1037
        	Error:      	Received unexpected error:
        	            	failed to load snapshot data:failed to migrate schema to version 22: failed to migrate up to version 22:ERROR 0012_candles_add_notional.sql: failed to run SQL migration: failed to execute SQL query "SELECT remove_continuous_aggregate_policy('trades_candle_5_minutes');\n": ERROR: deadlock detected (SQLSTATE 40P01)
        	Test:       	TestRestoreFromFullHistorySnapshotWithIndexesAndOrderTriggersAndProcessEvents
--- FAIL: TestRestoreFromFullHistorySnapshotWithIndexesAndOrderTriggersAndProcessEvents (3.90s)
=== RUN   TestRollingBackToHeightAcrossSchemaUpdateBoundary
2023-07-18T08:47:45.117Z	INFO	db-wipe-migrate	networkhistory/service_test.go:1242	Wiping database and migrating schema to version 22

### Expected behaviour

don't deadlock 

### Steps to reproduce

```bash
"
```


### Software version

develop

### Failing test

_No response_

### Jenkins run

_No response_

### Configuration used

_No response_

### Relevant log output

_No response_

--- a/datanode/sqlstore/migrations/0012_candles_add_notional.sql
+++ b/datanode/sqlstore/migrations/0012_candles_add_notional.sql
@@ -1,5 +1,8 @@
 -- +goose Up
 
+
+SELECT _timescaledb_internal.stop_background_workers();
+
 SELECT remove_continuous_aggregate_policy('trades_candle_1_minute');
 DROP MATERIALIZED VIEW IF EXISTS trades_candle_1_minute;
 
@@ -138,6 +141,8 @@ SELECT market_id,  vega_time as period_start,
 
 FROM trades
 GROUP BY market_id, vega_time;
+
+SELECT _timescaledb_internal.start_background_workers();
 
 -- +goose Down
 

